include/xtensor/xfile_array.hpp

```diff
@@ -548,29 +554,59 @@ namespace xt
         m_io_handler.configure_format(config);
     }
 
+    template <class E, class IOH>
+    inline void xfile_array_container<E, IOH>::ignore_empty_path(bool ignore)
+    {
+        m_ignore_empty_path = ignore;
+    }
+
+    template <class E, class IOH>
+    inline bool xfile_array_container<E, IOH>::enable_io(std::string& path)
+    {
+        bool res;
+        if (path.empty())
+        {
+            res = !m_ignore_empty_path;
+        }
+        else
+        {
+            res = true;
+        }
+        return res;
+    }
+
     template <class E, class IOH>
     inline void xfile_array_container<E, IOH>::set_path(std::string& path)
     {
         if (path != m_path)
         {
             // maybe write to old file
-            if (m_dirty && !m_path.empty())
+            if (m_dirty)
             {
-                m_io_handler.write(m_storage, m_path);
+                if (enable_io(m_path))
+                {
+                    m_io_handler.write(m_storage, m_path);
+                }
                 m_dirty = false;
             }
             m_path = path;
             // read new file
-            m_io_handler.read(m_storage, path);
+            if (enable_io(path))
+            {
+                m_io_handler.read(m_storage, path);
+            }
```

Comment:
Do we want to allow the user to explicitely pass an empty path?

Reply:
Maybe? An empty path with `ignore_empty_path(true)` effectively disable all IO, transforming the file-backed cached array to a regular array. It could have some applications...
But anyway, with `ignore_empty_path(false)` (default), the underlying IO handler will throw an exception, so we indirectly disallow empty paths by default.